PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/mysql/MySQLPropertyView.java

```diff
@@ -0,0 +1,269 @@
+import java.util.List;
+
+public class MySQLPropertyView extends BaseEditor implements MySQLPropertyMvpView {
+
+    public static final String ID = "com.microsoft.intellij.helpers.mysql.MySQLPropertyView";
+
+    private AzureHideableTitledSeparator overviewSeparator;
+    private MySQLPropertryOverviewPanel overview;
+    private AzureHideableTitledSeparator connectionSecuritySeparator;
+    private ConnectionSecurityPanel connectionSecurity;
+    private AzureHideableTitledSeparator connectionStringsSeparator;
+    private ConnectionStringsOutputPanel connectionStringsJDBC;
+    private ConnectionStringsOutputPanel connectionStringsSpring;
+    private JPanel rootPanel;
+    private JPanel contextPanel;
+    private JScrollPane scrollPane;
+    private MySQLPropertyActionPanel propertyActionPanel;
+    private DatabaseComboBox databaseComboBox;
+    private JLabel databaseLabel;
+    public static final String MYSQL_OUTPUT_TEXT_PATTERN_SPRING =
+            "spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver" + StringUtils.LF +
+                    "spring.datasource.url=jdbc:mysql://%s:3306/%s?useSSL=true&requireSSL=false" + StringUtils.LF +
+                    "spring.datasource.username=%s" + StringUtils.LF + "spring.datasource.password={your_password}";
+
+    public static final String MYSQL_OUTPUT_TEXT_PATTERN_JDBC =
+            "String url =\"jdbc:mysql://%s:3306/%s?useSSL=true&requireSSL=false\";" + StringUtils.LF +
+                    "myDbConn = DriverManager.getConnection(url, \"%s\", {your_password});";
+
+    private MySQLProperty property;
+
+    private Boolean originalAllowAccessToAzureServices;
+    private Boolean originalAllowAccessToLocal;
+
+    MySQLPropertyView() {
+        super();
+        overviewSeparator.addContentComponent(overview);
+        connectionSecuritySeparator.addContentComponent(connectionSecurity);
+        connectionStringsSeparator.addContentComponent(databaseLabel);
+        connectionStringsSeparator.addContentComponent(databaseComboBox);
+        connectionStringsSeparator.addContentComponent(connectionStringsJDBC);
+        connectionStringsSeparator.addContentComponent(connectionStringsSpring);
+        connectionStringsJDBC.getTitleLabel().setText("JDBC");
+        connectionStringsJDBC.getOutputTextArea().setText(getConnectionString(MYSQL_OUTPUT_TEXT_PATTERN_JDBC, null, null, null));
+        connectionStringsSpring.getTitleLabel().setText("Spring");
+        connectionStringsSpring.getOutputTextArea().setText(getConnectionString(MYSQL_OUTPUT_TEXT_PATTERN_SPRING, null, null, null));
+        init();
+        initListeners();
+    }
+
+    private String getConnectionString(final String pattern, final String hostname, final String database, final String username) {
+        if (StringUtils.isBlank(hostname) || StringUtils.isBlank(database) || StringUtils.isBlank(username)) {
+            return String.format(pattern, "{your_hostname}", "{your_database}", "{your_username}");
```

Comment:
create a default connection string constant and return it directly?

Reply:
the "pattern" parameter is already a pattern and have more than one (two) types currently. more than one constants seems not a good a idea.
these lines means set default values for hostname, database, and username.